File: BehaviorPattern/Template_Method/Python/Abstract_Template_Methods/Game.py
from abc import ABC, abstractmethod

class Game(ABC):
    # Template method
    def play(self):
        self.initialize()
        self.start_play()
        self.end_play()

    @abstractmethod
    def initialize(self):
        pass

    @abstractmethod
    def start_play(self):
        pass

    @abstractmethod
    def end_play(self):
        pass

File: BehaviorPattern/Template_Method/Golang/Abstract_Template_Methods/Game.go
package abstracttemplatemethods

import (

)

// Game interface defines the methods that need to be implemented by concrete classes
type Game interface {
	Initialize()
	StartPlay()
	EndPlay()
}

// play is the template method that defines the steps of the algorithm
func Play(g Game) {
	g.Initialize()
	g.StartPlay()
	g.EndPlay()
}